pkg/apis/flowcontrol/validation/validation.go

```diff
@@ -220,9 +216,17 @@ func ValidatePriorityLevelConfigurationSpec(spec *flowcontrol.PriorityLevelConfi
 		if spec.Queues <= 0 {
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("queues"), spec.Queues, "must be positive"))
 		}
-		if spec.HandSize < 0 {
+		if spec.HandSize <= 0 {
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("handSize"), spec.HandSize, "must be positive"))
 		}
+		if spec.HandSize > spec.Queues {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("handSize"), spec.HandSize,
+				fmt.Sprintf("should not be greater than queues (%d)", spec.Queues)))
+		}
+		if !shufflesharding.ValidateParameters(spec.Queues, spec.HandSize) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("handSize"), spec.HandSize,
+				fmt.Sprintf("must pass shuffle-sharding validation")))
```

Comment:
This error message is too generic.  The usual way to delegate checking and error message generation is for the invoked function to return a slice of error messages and then the code here packages each one into a `field.Invalid` call (empty slice means no errors).

Reply:
and the error message has to be user-facing. i.e. the content should tell the users "what is wrong and how to deal w/ that"

Reply:
It's hard to use generic validation from shufflesharding package here, let's keep both, one for flowcontrol(fairness and priority) and one for common users (who needs validation).

Reply:
Will revert this.

Reply:
I do not understand the problem.  Let there be just one validation function in the world, let it live here and return a slice of error messages.  What's wrong with that?